math/int_test.go

```diff
@@ -56,6 +56,20 @@ func (s *intTestSuite) TestNewIntFromBigInt() {
 	s.Require().NotEqual(r, i.BigInt())
 }
 
+func (s *intTestSuite) TestConvertToBigIntMutative() {
+	r := big.NewInt(42)
+	i := math.NewIntFromBigInt(r)
+
+	// Compare value of BigInt & BigIntMut
+	s.Require().Equal(i.BigInt(), i.BigIntMut())
+	
+	// Modify big.Int pointer and ensure i.BigIntMut change
```

Comment:
For comparison, can we add the same when using BigInt and prove it ins't mutated?